client/state/user-settings/thunks/save-two-step-sms-settings.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * External dependencies
+ */
+import wp from 'calypso/lib/wp';
+
+/**
+ * Internal dependencies
+ */
+import {
+	saveUserSettingsFailure,
+	saveUserSettingsSuccess,
+} from 'calypso/state/user-settings/actions';
+import { fromApi } from 'calypso/state/data-layer/wpcom/me/settings';
+import { USER_SETTINGS_SAVE } from 'calypso/state/action-types';
+
+const wpcom = wp.undocumented();
+
+const saveTwoStepSMSSettings = ( countryCode, phoneNumber ) => async ( dispatch ) => {
+	dispatch( { type: USER_SETTINGS_SAVE } );
+
+	const settings = {
+		two_step_sms_country: countryCode,
+		two_step_sms_phone_number: phoneNumber,
+	};
+
+	try {
+		const response = await wpcom.me().settings().update( settings );
+		dispatch( saveUserSettingsSuccess( fromApi( response ) ) );
+	} catch ( err ) {
+		dispatch( saveUserSettingsFailure( settings, err ) );
+		throw err;
```

Comment:
Nice work chaining the errors all the way up here.

However, it's not very usual for an action thunk to purposely throw an error; should we add a docblock and explain the use case when it's expected for an error to be thrown? I think it would make sense, because we often don't wrap action dispatching in a `try/catch` unless there is a good reason for that.

Reply:
Yup added in https://github.com/Automattic/wp-calypso/pull/49000/commits/44cfb0d5e2ce1cd3cf9ef50a9331fcc3cdfb285d